### DISCUSSION TITLE
React Native, fazer cache com banco local ou não?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1604] [Perguntas e Respostas]
Galera,
com react-native, estou estudando, mas me surgiu uma dúvida, quando usar cache ou banco local no dispositivo?

Por exemplo, vou fazer um app de anúncios para brincar, mas deixo o app sem nenhum cachê ou dado local? Baixo as imagens e salvo em um cachê no dispositivo? Ou salvo os itens buscados no dispositivo também?

Como vocês fazem?

----------------
:relaxed:  
by: Wilson Neto  
[Linkedin](https://linkedin.com/in/wilsonnetobr/)
[Blog](http://wilsonneto.com.br)
[Github](https://github.com/wilsonneto-dev)

**Comment by eliseumds** [comment]
Essa é uma excelente pergunta e uma das mais difíceis de responder da indústria, porque é extremamente dependente da situação. O mesmo rola com CDNs, com bancos de dados, com sessões de usuários, etc. Qualquer que sejam suas escolhas, crie dispositivos pra poder limpar o cache facilmente e sempre defina limites de items/espaço (LRU é o mais comum) e expiration date. É bem comum cachear imagens em mobile apps.
